part2_server.py

```diff
@@ -1,43 +1,33 @@
 # Group#: G37
 # Student Names: Ryley McRae & Niko Andrianos
 
+from socket import *
+import random, time
+
 """
-    This program implements a server for a UDP Pinger 
-    application on a local host.
+    Implements a server for a UDP Pinger application on a local host.
 """
+class Server:
+    def __init__(self, host: str = "127.0.0.1", port: int = 12000, bufferSize: int = 2048):
+        self.host = host
+        self.port = port
+        self.bufferSize = bufferSize
+        self.serverSocket = socket(AF_INET, SOCK_DGRAM)
+
+    def run(self):
+        self.serverSocket.bind((self.host, self.port))
+        print(f"The server is listening on port {self.port}")
+        while True:
+            message, clientAddress = self.serverSocket.recvfrom(self.bufferSize)
+            ignoreMessageFlag = (random.random() < 0.1) # create a flag with a 10% probability to simulate packet loss
+            if ignoreMessageFlag:
+                continue
+            else:
+                responseMessage = ' '.join(message.decode().split(' ')[:3]) + " ditto" # perform string manipulation to the desired format
```

Comment:
I prefer:
        decodedMessageList = message.decode().split(" ")
        responseMessage = f"{decodedMessageList[0]} {decodedMessageList[1]} - ditto"
for readability, but I can respect moving to one line is fine too.

Reply:
1 or 2 lines works. I was wondering if we should implement a way of telling the server what the "splitter" is (aka the symbol the splits the message identifier "PING #" from the message content "Hello world!") so that we may pass messages of arbitrary formats

Reply:
We could add something to this. In the most recent commit, I didn't add this.